dlt/pipeline/pipeline.py

```diff
@@ -445,7 +445,12 @@ def extract(
                 )
                 # commit load packages with state
                 extract_step.commit_packages()
-                return self._get_step_info(extract_step)
+
+                info = self._get_step_info(extract_step)
+                state = self._container[StateInjectableContext].state
+                state["_local"]["_last_extracted_count"] = info.total_rows_count
```

Comment:
Figuring out if it's possible to do this in a less hacky way without breaking anything